Makefile

```diff
@@ -70,15 +85,25 @@ doc/nitc/index.html: bin/nitdoc bin/nitls
 		--piwik-tracker "pratchett.info.uqam.ca/piwik/" \
 		--piwik-site-id "3"
 
-man:
-	$(MAKE) -C share/man
+.PHONY: man $(PROJECT_MANDIR)
+man $(PROJECT_MANDIR):
+	$(MAKE) -C $(PROJECT_MANDIR)
 
-clean:
-	rm -rf -- doc/stdlib doc/nitc || true
-	cd c_src; make clean
-	cd src; make clean
-	cd tests; make clean
-	cd share/man; make clean
+# `clean` `distclean` and `mostlyclean` avoid deleting files usually provided by
+# the maintainers. `mostlyclean` also avoids deleting Nit tools in `./bin`.
+# `maintainer-clean` deletes all files that can be regenerated by this Makefile.
+#
+# WARNING: The `maintainer-clean` target is intended for maintainers to use; it
+# deletes files that may need special tools to rebuild.
+#
+# See also: https://www.gnu.org/prep/standards/html_node/Standard-Targets.html
+.PHONY: clean distclean mostlyclean maintainer-clean
+clean distclean mostlyclean maintainer-clean:
+	rm -rf -- $(PROJECT_DOCDIR)/stdlib $(PROJECT_DOCDIR)/nitc || true
+	$(MAKE) -C ./c_src clean || true
+	$(MAKE) -C ./src $@
+	$(MAKE) -C ./tests $@
+	$(MAKE) -C ./share/man $@
 	for m in $(PROGS); do \
-		$(MAKE) clean -C "$$m"; \
-		done || true
```

Comment:
You can drop the commit "fixing" the indentation since remove the line altogether afterward.